adoc/extensions/sycl_khr_free_function_commands.adoc

```diff
@@ -0,0 +1,928 @@
+                           Args&&... args); (6)
+
+}
+----
+_Effects_: Enqueues a kernel object that is invoked for every work-item in the
+specified [code]#sycl::range#, where each work-item contributes to one or more
+reductions.
+Work-items are grouped into work-groups of size [code]#size#.
+The arguments in [code]#args# are passed to the kernel in the same order.
+
+'''
+
+.[apititle]#launch_task# (kernel function)
+[source,role=synopsis,id=api:launch_task]
+----
+namespace sycl::khr {
+
+template <typename KernelType>
+void launch_task(sycl::handler& h, const KernelType& k); (1)
+
+template <typename KernelType>
+void launch_task(sycl::queue q, const KernelType& k);    (2)
+
+}
+----
+_Effects (1)_: Equivalent to [code]#h.single_task(k)#.
+
+_Effects (2)_: Equivalent to [code]#h.submit([&](handler& h) { launch_task(h,
+k); })#.
+
+'''
+
+.[apititle]#launch_task# (kernel object)
+[source,role=synopsis,id=api:launch_task-kernel]
+----
+namespace sycl::khr {
+
+template <typename Args...>
+void launch_task(sycl::queue q, const sycl::kernel& k, Args&&... args);    (1)
+
+template <typename Args...>
+void launch_task(sycl::handler& h, const sycl::kernel& k, Args&&... args); (2)
+
+}
+----
+_Effects_: Enqueues a kernel object as a device task.
+The arguments in [code]#args# are passed to the kernel in the same order.
+
+'''
+
+=== Memory operations
+
+.[apititle]#memcpy#
+[source,role=synopsis,id=api:memcpy]
+----
+namespace sycl::khr {
+
+void memcpy(sycl::handler& h, void* dest, const void* src, size_t numBytes); (1)
+
+void memcpy(sycl::queue q, void* dest, const void* src, size_t numBytes);    (2)
+
+}
+----
+_Effects (1)_: Equivalent to [code]#h.memcpy(dest, src, numBytes)#.
+
+_Effects (2)_: Equivalent to [code]#q.submit([&](handler& h) { memcpy(h, dest,
+src, numBytes); })#.
+
+'''
+
+.[apititle]#copy# (USM pointers)
+[source,role=synopsis,id=api:copy-pointer]
+----
+namespace sycl::khr {
+
+template <typename T>
+void copy(sycl::handler& h, const T* src, T* dest, size_t count); (1)
+
+template <typename T>
+void copy(sycl::queue q, const T* src, T* dest, size_t count);    (2)
+
+}
+----
+
+Copies between two USM pointers.
+
+_Constraints_: [code]#T# must be <<device-copyable>>.
+
+_Preconditions_: [code]#src# and [code]#dest# must be host pointers or USM
+pointers accessible on the device.
+[code]#src# and [code]#dest# must point to allocations of at least [code]#count#
+elements of type [code]#T#.
+
+_Effects (1)_: Equivalent to [code]#h.copy(src, dest, count)#.
+
+_Effects (2)_: Equivalent to [code]#q.submit([&](handler& h) { copy(h, src,
+dest, count); })#
+
+'''
+
+.[apititle]#copy# (accessors, host to device)
+[source,role=synopsis,id=api:copy-accessor-h2d]
+----
+namespace sycl::khr {
+
+template <typename SrcT, typename DestT, int DestDims, access_mode DestMode>
+void copy(sycl::handler& h,
+          const SrcT* src,
+          sycl::accessor<DestT, DestDims, DestMode, target::device> dest); (3)
```

Comment:
```suggestion
          sycl::accessor<DestT, DestDims, DestMode, target::device> dest); (1)
```

Our style is to restart the numbering from (1) in each synopsis code block.

Reply:
Even when the code block is effectively a continuation of the one before?  These are all overloads of `copy`, and I only split them up in an attempt to keep things readable.

I can imagine there might be cases in future where we might want to say something like "Overload (3) of `copy`", and that seems simpler than saying "Overload (1) of `copy` in the second synopsis code block".

I don't feel strongly about this, I just wanted to make sure you saw that these are all `copy` overloads and talk that through before making the change.

Reply:
If you feel that the APIs are so closely related that the numbering should be a continuation, then I would encourage you to list them all in the same synopsis.  The advantage of this is that there will be a single `[apidef]` for all the `sycl::copy` functions, and any occurrence of `[api]#copy#` will hyperlink to that synopsis.  By splitting the `copy` functions into separate synopses, you lose this automatic hyperlinking feature.

The style I have been gravitating to is to prefer a single synopsis for all overloads of a function.  For example, in the reformatted `queue` section, there is a single synopsis for all 8 overloads of `queue::copy`.

Reply:
I feel that they're related in that they're all overloads of the same function, but I think that there's enough dissimilarity in the overloads that splitting them up improves readability.  There are 12 overloads in total here, but it's really 4 groups of related overloads.

I've restarted the numbering at 1, as you originally suggested: 88b540aff2c79f9c2fba8b2cf0bfa917b39434dc.  Everything is now consistent within this KHR.